src/Step.cc

```diff
@@ -283,6 +290,11 @@ std::string to_string(Step::Type type)
     return "unknown";
 }
 
+bool executes_script(Step::Type step_type) noexcept
```

Comment:
I'd be careful with the `noexcept`... are we absolutely sure that the `find()` can never throw?

Reply:
Hmm, since the [`Compare`](https://timsong-cpp.github.io/cppwp/n3337/associative.reqmts.except) object in [`std::set::find`](https://en.cppreference.com/w/cpp/container/set/find) can throw there is no guarantee about `noexcept`. Since we use an `enum` I would __assume__  that this is converted to an `int` and I would be very surprised if this will throw in the `Compare` object. But at least you are right. To be on the safe side I remove it.